unit_tests/RecordUnitTests.d

```diff
@@ -524,3 +524,25 @@ D");
 
 }
 
+@("issue #foo - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
```

Comment:
You will benefit from this when it propagates to GitHub. GitHub uses the TextMate bundle to highlight D code.